base/lnls-prune-artifacts.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+set -Eeu
+
+# Filter out from the $1 list of paths any exact match of a parent directory
+# from any path in the $2 exclude list.
+#
+# Both list are treated as newline-separated strings, and must consist
+# exclusively of absolute paths.
+filter_out_paths() {
+    list="$1"
+    exclude_list=$(echo "$2" | grep -v '^$')
```

Comment:
I think if we accept a list with empty lines for second argument, we should try to do the same for the first one too.

Reply:
Fair enough. I've also included a mention in the docstring.